Blog/GamingGuruBlog.Web/GamingGuruBlog.Domain/Interfaces/IBlogServices.cs

```diff
@@ -6,13 +6,27 @@ namespace GamingGuruBlog.Domain.Interfaces
     public interface IBlogServices
     {
         List<Category> GetAllCategories();
+        List<Category> GetUsedCategories();
         int AddNewBlogPost(BlogPost newPost);
         List<Category> GetAssignedCategories(int blogID);
         void AddCategoriesToBlogPost(int blogPostID, List<Category> categoryIDs);
         List<Tag> AddCreatedTags(List<Tag> tagNames);
         void AddTagsToBlog(int blogID, List<Tag> tagID);
         BlogPost GetBlogPost(int blogID);
+        BlogPost GetApprovedBlogPost(int id);
         List<Tag> GetAllTags();
         void ProcessEditedBlogPost(BlogPost editedBlogPost);
+        List<BlogPost> GetBlogPostByCategoryID(int categoryID);
+        List<BlogPost> AllBlogPostsByTag(int tagID);
+        void DeleteBlogPost(int blogID);
+        List<BlogPost> GetAllBlogPosts();
+        List<BlogPost> GetApprovedBlogPosts();
+        User GetUser(string userID);
+        void EditUser(User editedUser);
+        List<User> GetAllUsers();
+        void DeleteCategory(int categoryID);
```

Comment:
This service seems a bit big, you might want to break it up...especially since Category is a separate entity from BlogPost (even though they're related).